src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -113,167 +115,189 @@ struct GradientVariablesEntropy end
+                                                                                       GradientVariablesPrimitive
+                                                                                       })
+    return gradient
 end
 
 # the first argument is always the "transformed" variables.
 @inline function convert_derivative_to_primitive(w, gradient_entropy_vars,
-                                                 equations::CompressibleNavierStokesDiffusion2D{GradientVariablesEntropy})
-
-  # TODO: parabolic. This is inefficient to pass in transformed variables but then transform them back.
-  # We can fix this if we directly compute v1, v2, T from the entropy variables
-  u = entropy2cons(w, equations) # calls a "modified" entropy2cons defined for CompressibleNavierStokesDiffusion2D
-  rho, rho_v1, rho_v2, _ = u
-
-  v1 = rho_v1 / rho
-  v2 = rho_v2 / rho
-  T  = temperature(u, equations)
-
-  return SVector(gradient_entropy_vars[1],
-                 T * (gradient_entropy_vars[2] + v1 * gradient_entropy_vars[4]), # grad(u) = T*(grad(w_2)+v1*grad(w_4))
-                 T * (gradient_entropy_vars[3] + v2 * gradient_entropy_vars[4]), # grad(v) = T*(grad(w_3)+v2*grad(w_4))
-                 T * T * gradient_entropy_vars[4]                                # grad(T) = T^2*grad(w_4))
-                )
+                                                 equations::CompressibleNavierStokesDiffusion2D{
+                                                                                                GradientVariablesEntropy
+                                                                                                })
+
+    # TODO: parabolic. This is inefficient to pass in transformed variables but then transform them back.
+    # We can fix this if we directly compute v1, v2, T from the entropy variables
+    u = entropy2cons(w, equations) # calls a "modified" entropy2cons defined for CompressibleNavierStokesDiffusion2D
+    rho, rho_v1, rho_v2, _ = u
+
+    v1 = rho_v1 / rho
+    v2 = rho_v2 / rho
+    T = temperature(u, equations)
+
+    return SVector(gradient_entropy_vars[1],
+                   T * (gradient_entropy_vars[2] + v1 * gradient_entropy_vars[4]), # grad(u) = T*(grad(w_2)+v1*grad(w_4))
+                   T * (gradient_entropy_vars[3] + v2 * gradient_entropy_vars[4]), # grad(v) = T*(grad(w_3)+v2*grad(w_4))
+                   T * T * gradient_entropy_vars[4])
 end
 
-
 # This routine is required because `prim2cons` is called in `initial_condition`, which
 # is called with `equations::CompressibleEulerEquations2D`. This means it is inconsistent
 # with `cons2prim(..., ::CompressibleNavierStokesDiffusion2D)` as defined above.
 # TODO: parabolic. Is there a way to clean this up?
-@inline prim2cons(u, equations::CompressibleNavierStokesDiffusion2D) =
+@inline function prim2cons(u, equations::CompressibleNavierStokesDiffusion2D)
     prim2cons(u, equations.equations_hyperbolic)
-
+end
```

Comment:
I'd prefer not adding `function` blocks when we don't have to, but not a big deal.

Reply:
This is done automatically if a short function style is used in the code but now the line is too long. E.g., something like
```julia
foo(bar) = bar^2
```
will not be reformatted.